tests/e2e/standalone/workflow_test.go

```diff
@@ -0,0 +1,739 @@
+				"--instance-id=purge-all-"+strconv.Itoa(i))
+			require.NoError(t, err, output)
+			_, _ = cmdWorkflowTerminate(appID, "purge-all-"+strconv.Itoa(i))
+		}
+
+		output, err := cmdWorkflowPurge(appID, redisConnString, "--all")
+		require.NoError(t, err, output)
+		assert.Contains(t, output, `Purged workflow instance "purge-all-1"`)
+		assert.Contains(t, output, `Purged workflow instance "purge-all-0"`)
+
+		output, err = cmdWorkflowList(appID, "-o", "json", redisConnString)
+		require.NoError(t, err)
+		assert.NotContains(t, output, "purge-all-0")
+		assert.NotContains(t, output, "purge-all-1")
+	})
+
+	t.Run("purge older than duration", func(t *testing.T) {
+		output, err := cmdWorkflowRun(appID, "SimpleWorkflow",
+			"--instance-id=purge-older")
+		require.NoError(t, err)
+
+		time.Sleep(5 * time.Second)
+
+		output, err = cmdWorkflowPurge(appID, redisConnString, "--all-older-than", "1s")
+		require.NoError(t, err, output)
+		assert.Contains(t, output, "Purging 1 workflow instance(s)")
+		assert.Contains(t, output, `Purged workflow instance "purge-older"`)
+
+		output, err = cmdWorkflowList(appID, "-o", "json", redisConnString)
+		require.NoError(t, err, output)
+		assert.NotContains(t, output, "purge-older")
+	})
+
+	t.Run("also purge scheduler", func(t *testing.T) {
+		output, err := cmdWorkflowRun(appID, "EventWorkflow",
+			"--instance-id=also-sched")
+		require.NoError(t, err)
+
+		output, err = cmdWorkflowTerminate(appID, "also-sched")
+		require.NoError(t, err, output)
+
+		output, err = cmdSchedulerList()
+		require.NoError(t, err)
+		assert.Greater(t, len(strings.Split(output, "\n")), 2)
+
+		output, err = cmdWorkflowPurge(appID, "also-sched")
+		require.NoError(t, err, output)
+
+		output, err = cmdSchedulerList()
+		require.NoError(t, err)
+		assert.Len(t, strings.Split(output, "\n"), 2)
+	})
+}
+
+func TestWorkflowFilters(t *testing.T) {
+	if isSlimMode() {
+		t.Skip("skipping workflow tests in slim mode")
+	}
+
+	cmdUninstall()
+	ensureDaprInstallation(t)
+	t.Cleanup(func() {
+		must(t, cmdUninstall, "failed to uninstall Dapr")
+	})
+
+	runFilePath := "../testdata/run-template-files/test-workflow.yaml"
+	appID := "test-workflow"
+	t.Cleanup(func() {
+		cmdStopWithAppID(appID)
+		waitAppsToBeStopped()
+	})
+	args := []string{"-f", runFilePath}
+
+	go func() {
+		o, _ := cmdRun("", args...)
+		t.Log(o)
+	}()
+
+	time.Sleep(5 * time.Second)
+
+	_, _ = cmdWorkflowRun(appID, "SimpleWorkflow", "--instance-id=simple-1")
+	_, _ = cmdWorkflowRun(appID, "LongWorkflow", "--instance-id=long-1")
+	output, err := cmdWorkflowRun(appID, "EventWorkflow", "--instance-id=suspend-test")
+	require.NoError(t, err, output)
+
+	time.Sleep(2 * time.Second)
+	_, _ = cmdWorkflowSuspend(appID, "suspend-test")
+
+	t.Run("filter by status", func(t *testing.T) {
+		output, err := cmdWorkflowList(appID, redisConnString, "--filter-status", "SUSPENDED")
+		require.NoError(t, err)
+		assert.Contains(t, output, "suspend-test")
```

Comment:
I think it'd be good to also check it doesn't contain `simple-1` and `long-1`. Same applies for the rest of the filters.